docs/DeveloperGuide.adoc

```diff
@@ -153,6 +153,104 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+// end::followup[]
+
+[[profileImplementation]]
+// tag::profile[]
+=== Profile feature
+==== Implementation
+
+The profile feature allows the user to be able to see the full details of the patient as well as all relevant visit information (See <<visitImplementation, Visit Implementation>>) from a single panel. Attributes of the patient is extracted from the `Person` instance, including the `Name`, `Address`, `Phone`, `Email`, `Tags` and `VisitList`, which can be easily extended following the implementation detailed below. The patient profile feature is facilitated by `ProfileWindow`, which is instanced and instantly hidden (`hide()`) upon the instancing of `MainWindow` on VISIT's start up. +
+ +
+Given below is a simple example usage scenario demonstrate how Profile feature behaves and interacts with the other components.
+
+Step 1. The user executes `profile 3` command view the profile of the 3rd person shown in VISIT's current list. It extracts the `Person` instance from the filtered Person list from the current `model` and returns a `CommandResult` with the `Person` instance as an attribute. `MainWindow#executeCommand()` is called and the check for `CommandResult#isShowProfile()` passes, calling `ProfileWindow#setup()` which populates the hidden Profile Window with the attributes of the `Person` instance before `MainWindow#handleProfilePanel()` is called to show the Profile Window.
+
+image::Profile0.png[]
+_{Show activity diagram of opening profile window}_
+
+Step 2. After viewing the relevant information, the user presses the 'esc' key which closes the form. This causes an event handler in `ProfileWindow` which listens for a `KeyEvent.KEY_RELEASED` event where the event code equates to `KeyCode.ESCAPE` ('esc' key) to trigger, which hides the profile window.
+
+image::Profile1.png[]
+_{Show activity diagram of closing profile window}_
+
+[NOTE]
+The user is able to close the `ProfileWindow` form by clicking the 'x' on the top right of the form as well.
+
+This sequence diagram summarizes what happens when a user executes the Profile Command:
+
+image::ProfileSequenceDiagram.png[]
+_{Show extended sequence diagram of opening and closing profile window}_
+
+==== Design Considerations
+
+===== Aspect: How Profile shows the information from the patient
+
+* **Alternative 1 (current choice):** Does not create an explicit `Profile` instance.
+** Pros: Easy to implement, more straight forward (Just show data without the need to store what is essentially a clone of a `Person` instance).
+** Cons: May cause dependency issues as this increases coupling, with relation to the Profile and Patient class. Foreseeably, this may cause issues especially when patient attributes are to be added or removed.
+* **Alternative 2:** Create an explicit `Profile` instance every call.
+** Pros: Decreases coupling by cutting down on the direct calls to the `Person` instance.
+** Cons: We must ensure that the `Profile` instance generated is correct every time.
```

Comment:
I think that you guys should try to elaborate on your current choice, try to go beyond listing the pros and cons and explain why you have chosen the first alternative over the other 😄